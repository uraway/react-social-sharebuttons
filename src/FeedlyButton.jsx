import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class FeedlyButton extends Component {
  constructor(props) {
    super(props);
    this.state = ({
      layout: '',
      width: '56px',
      height: '131px',
    });
  }

  componentDidMount() {
    this.layoutSet();
  }

  layoutSet() {
    if (this.props.layout === 'rectangle-volume-big') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-volume-big_2x.png',
        height: '56px',
        width: '131px',
      });
    } else if (this.props.layout === 'rectangle-volume-medium') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-volume-medium_2x.png',
        height: '28px',
        width: '71px',
      });
    } else if (this.props.layout === 'rectangle-volume-small') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-volume-small_2x.png',
        height: '20px',
        width: '66px',
      });
    } else if (this.props.layout === 'rectangle-flat-big') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-flat-big_2x.png',
        height: '28px',
        width: '71px',
      });
    } else if (this.props.layout === 'rectangle-flat-medium') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-flat-medium_2x.png',
        height: '28px',
        width: '71px',
      });
    } else if (this.props.layout === 'rectangle-flat-small') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-flat-small_2x.png',
        height: '20px',
        width: '66px',
      });
    } else if (this.props.layout === 'square-volume') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-square-volume_2x.png',
        height: '28px',
        width: '28px',
      });
    } else if (this.props.layout === 'flat-green') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-square-flat-green_2x.png',
        height: '28px',
        width: '28px',
      });
    } else if (this.props.layout === 'circle-flat-green') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-circle-flat-green_2x.png',
        height: '28px',
        width: '28px',
      });
    } else if (this.props.layout === 'logo-green') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-logo-green_2x.png',
        height: '28px',
        width: '28px',
      });
    } else if (this.props.layout === 'square-flat-black') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-square-flat-black_2x.png',
        height: '28px',
        width: '28px',
      });
    } else if (this.props.layout === 'circle-flat-black') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-circle-flat-black_2x.png',
        height: '28px',
        width: '28px',
      });
    } else if (this.props.layout === 'logo-black') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-logo-black_2x.png',
        height: '28px',
        width: '28px',
      });
    } else if (this.props.layout === 'square-flat-white') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-square-flat-white_2x.png',
        height: '28px',
        width: '28px',
      });
    } else if (this.props.layout === 'circle-flat-white') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-circle-flat-white_2x.png',
        height: '28px',
        width: '28px',
      });
    } else if (this.props.layout === 'logo-white') {
      this.setState({
        layout: 'http://s3.feedly.com/img/follows/feedly-follow-logo-white_2x.png',
        height: '28px',
        width: '28px',
      });
    }
  }

  render() {
    return (
      <a href={`http://cloud.feedly.com/#subscription%2Ffeed%2F${encodeURIComponent(this.props.feedurl)}`} target="blank">
        <img src={this.state.layout} alt="follow us in feedly" style={{ width: this.state.width, height: this.state.height }} />
      </a>
    );
  }
}

FeedlyButton.propTypes = {
  feedurl: PropTypes.string.isRequired,
  layout: PropTypes.string,
};

FeedlyButton.defaultProps = {
  layout: 'rectangle-volume-small'
};

/*
encoded
http://cloud.feedly.com/#subscription%2Ffeed%2Fhttp%3A%2F%2Furaway.hatenablog.com%2Ffeed
http://cloud.feedly.com/#subscription%2Ffeed%2Fhttp%3A%2F%2Furaway.hatenablog.com%2Ffeed
decoded
http://cloud.feedly.com/#subscription/feed/http://uraway.hatenablog.com/feed
*/
