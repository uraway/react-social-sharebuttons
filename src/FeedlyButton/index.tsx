type Layout =
  | 'rectangle-volume-big'
  | 'rectangle-volume-medium'
  | 'rectangle-volume-small'
  | 'rectangle-flat-big'
  | 'rectangle-flat-medium'
  | 'rectangle-flat-small'
  | 'square-volume'
  | 'flat-green'
  | 'circle-flat-green'
  | 'logo-green'
  | 'square-flat-black'
  | 'circle-flat-black'
  | 'logo-black'
  | 'square-flat-white'
  | 'circle-flat-white'
  | 'logo-white';

type Props = {
  feedurl: string;
  layout?: Layout;
  width?: string;
  height?: string;
};

export default function FeedlyButton({
  feedurl,
  layout = 'rectangle-volume-big',
  width,
  height,
  ...others
}: Props): JSX.Element {
  const layoutSet: { [key in Layout]: { width: string; height: string; img: string } } = {
    'rectangle-volume-big': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-volume-big_2x.png',
      height: '56px',
      width: '131px',
    },
    'rectangle-volume-medium': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-volume-medium_2x.png',
      height: '28px',
      width: '71px',
    },
    'rectangle-volume-small': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-volume-small_2x.png',
      height: '20px',
      width: '66px',
    },
    'rectangle-flat-big': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-flat-big_2x.png',
      height: '28px',
      width: '71px',
    },
    'rectangle-flat-medium': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-flat-medium_2x.png',
      height: '28px',
      width: '71px',
    },
    'rectangle-flat-small': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-rectangle-flat-small_2x.png',
      height: '20px',
      width: '66px',
    },
    'square-volume': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-square-volume_2x.png',
      height: '28px',
      width: '28px',
    },
    'flat-green': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-square-flat-green_2x.png',
      height: '28px',
      width: '28px',
    },
    'circle-flat-green': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-circle-flat-green_2x.png',
      height: '28px',
      width: '28px',
    },
    'logo-green': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-logo-green_2x.png',
      height: '28px',
      width: '28px',
    },
    'square-flat-black': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-square-flat-black_2x.png',
      height: '28px',
      width: '28px',
    },
    'circle-flat-black': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-circle-flat-black_2x.png',
      height: '28px',
      width: '28px',
    },
    'logo-black': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-logo-black_2x.png',
      height: '28px',
      width: '28px',
    },
    'square-flat-white': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-square-flat-white_2x.png',
      height: '28px',
      width: '28px',
    },
    'circle-flat-white': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-circle-flat-white_2x.png',
      height: '28px',
      width: '28px',
    },
    'logo-white': {
      img: 'http://s3.feedly.com/img/follows/feedly-follow-logo-white_2x.png',
      height: '28px',
      width: '28px',
    },
  };
  const currentLayout = layoutSet[layout];
  return (
    <a href={`https://feedly.com/i/subscription/feed/${encodeURIComponent(feedurl)}`} target="blank">
      <img
        src={currentLayout.img}
        alt="follow us in feedly"
        style={{ width: width || currentLayout.width, height: height || currentLayout.height }}
        {...others}
      />
    </a>
  );
}

FeedlyButton.defaultProps = {
  layout: 'rectangle-volume-small',
  width: '',
  height: '',
};
