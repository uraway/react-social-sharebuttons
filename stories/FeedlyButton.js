// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import { FeedlyButton } from '../dist/index.esm.js';

const layoutOptions = {
  'rectangle-volume-big': 'rectangle-volume-big',
  'rectangle-volume-medium': 'rectangle-volume-medium',
  'rectangle-volume-small': 'rectangle-volume-small',
  'rectangle-flat-big': 'rectangle-flat-big',
  'rectangle-flat-medium': 'rectangle-flat-medium',
  'rectangle-flat-small': 'rectangle-flat-small',
  'square-volume': 'square-volume',
  'flat-green': 'flat-green',
  'circle-flat-green': 'circle-flat-green',
  'logo-green': 'logo-green',
  'square-flat-black': 'square-flat-black',
  'circle-flat-black': 'circle-flat-black',
  'logo-black': 'logo-black',
  'square-flat-white': 'square-flat-white',
  'circle-flat-white': 'circle-flat-white',
  'logo-white': 'logo-white',
};

export default storiesOf('FeedlyButton', module)
  .addDecorator(withKnobs)
  .add('configured', () => (
    <FeedlyButton feedurl={text('feedurl', '')} layout={select('layout', layoutOptions, 'rectangle-volume-big')} />
  ));
