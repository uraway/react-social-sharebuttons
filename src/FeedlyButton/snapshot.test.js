import React from 'react';
import renderer from 'react-test-renderer';
import FeedlyButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<FeedlyButton feedurl="url" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer.create(<FeedlyButton url="url" layout="square-flat-white" />).toJSON();
  expect(tree).toMatchSnapshot();
});
