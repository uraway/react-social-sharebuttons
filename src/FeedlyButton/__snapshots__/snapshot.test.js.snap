// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<a
  href="https://feedly.com/i/subscription/feed/undefined"
  target="blank"
>
  <img
    alt="follow us in feedly"
    src="http://s3.feedly.com/img/follows/feedly-follow-square-flat-white_2x.png"
    style={
      Object {
        "height": "28px",
        "width": "28px",
      }
    }
    url="url"
  />
</a>
`;

exports[`renders correctly with required props 1`] = `
<a
  href="https://feedly.com/i/subscription/feed/url"
  target="blank"
>
  <img
    alt="follow us in feedly"
    src="http://s3.feedly.com/img/follows/feedly-follow-rectangle-volume-small_2x.png"
    style={
      Object {
        "height": "20px",
        "width": "66px",
      }
    }
  />
</a>
`;
